### PR TITLE
hp98x5: Filesystem driver for HP9825/HP9845 added

### PR DESCRIPTION
**Pull request by fulivi**
Hi,
I'm opening this PR for the FS driver I've just added to floptool. This driver handles the disk filesystems of HP9825 and HP9845 machines.
Thanks.
--F.Ulivi

### CODE REVIEW

File: src/devices/bus/ieee488/hp9895.cpp [hunk 1]
```diff
@@ -57,6 +57,7 @@
 #include "emu.h"
 #include "hp9895.h"
 #include "formats/hpi_dsk.h"
+#include "formats/fs_hp98x5.h"
```
Comment (cuavas) [inline src/devices/bus/ieee488/hp9895.cpp:60]: Please put the headers from `lib/formats` in alphabetical order here.

File: src/lib/formats/fsmeta.cpp [hunk 2]
```diff
@@ -31,6 +31,7 @@ const char *meta_data::entry_name(meta_name name)
 	case meta_name::owner_id: return "owner_id";
 	case meta_name::attributes: return "attributes";
 	case meta_name::oem_name: return "oem_name";
+		case meta_name::rec_size: return "rec_size";
```
Comment (cuavas) [inline src/lib/formats/fsmeta.cpp:34]: Please be consistent with indentation – this file uses outdented case labels.

**Review by cuavas — COMMENTED** [review]
Can you please explain why you need to add an extra filesystem metadata type, and what it’s used for?  This is the kind of thing that needs to be thought through properly.

**Comment by fulivi** [comment]
The reason I added a new metadata is simple: HP9845 files have a "logical record size". This is a 1st class metadata, it's stored in directory entries alongside file name, file size, etc. I don't think that anything that's currently in `fsmeta.h` approximates record size very well. So I had to add it. Do you think I could have done this in some alternative way, please?

**Comment by fulivi** [comment]
@galibert May I ask you to review this PR, please?
Thanks a lot.

File: src/devices/bus/hp9845_io/hp9885.cpp [hunk 3]
```diff
@@ -147,6 +147,7 @@
 #include "emu.h"
 #include "hp9885.h"
 #include "formats/hpi_dsk.h"
+#include "formats/fs_hp98x5.h"
```
Comment (cuavas) [inline src/devices/bus/hp9845_io/hp9885.cpp:150]: These ones are still out-of-order.

**Comment by cuavas** [comment]
> The reason I added a new metadata is simple: HP9845 files have a "logical record size". This is a 1st class metadata, it's stored in directory entries alongside file name, file size, etc. I don't think that anything that's currently in `fsmeta.h` approximates record size very well. So I had to add it. Do you think I could have done this in some alternative way, please?

I don’t think it’s going to scale out well if we need to add one of these for every piece of first class filesystem metadata.  I mean, Macintosh HFS has a colour label and icon position as first-class filesystem meta, but I don’t see those in the list.

**Comment by fulivi** [comment]
Ok, I think I see your point. I'll keep the rec_size metadata off-tree as it's useful for me.

**Comment by fulivi** [comment]
Would it be more acceptable to show record size in the `attributes` metadata? As I understand it, `attributes` report all those aspects of the files that don't fit in other metadata because they are very filesystem-specific. For example, I used this metadata in ISIS FS for the various "write-only", "hidden", etc. flags.
Thanks.

**Comment by fulivi** [comment]
So come on and let me know should I stay or should I go?

**Comment by fulivi** [comment]
So long..

**Comment by galibert** [comment]
Sorry, I had forgotten about it, drowing in 68k.  Still want it in?

**Pull request closed without merging**